src/core/settings/qgssettingsentry.cpp

```diff
@@ -151,6 +151,7 @@ bool QgsSettingsEntryBase::setVariantValue( const QVariant &value, const QString
 
 bool QgsSettingsEntryBase::setVariantValue( const QVariant &value, const QStringList &dynamicKeyPartList ) const
 {
+  mHasChanged = true;
```

Comment:
@nyalldawson why haven't you moved this down after the check `if ( value != currentValue )`

Reply:
I can't recall exactly. Possibly it was to avoid the overhead of retrieving the current value. But happy for you to move it!